my-app/src/app/modules/courses-page/courses-page.component.ts

```diff
@@ -27,14 +29,16 @@ export class CoursesPageComponent implements OnInit {
   }
 
   public onDeleteCourse(courseId: number): void {
-    const courseTitle: string | undefined = this.coursesHandlerService.getItemById(courseId)?.title;
-
-    if (confirm(
-      `Are you sure to delete '${courseTitle}' course`) && this.visibleCourses
-    ) {
-      this.visibleCourses = this.coursesHandlerService.removeItem(courseId);
-      console.log('Course has been deleted successfully')
-    }
+    const foundCourse: ICourse | undefined = this.coursesHandlerService.getItemById(courseId);
+
+    if (foundCourse) {
+      if (confirm(
+        `Are you sure to delete '${foundCourse.title}' course`) && this.visibleCourses
+      ) {
+        this.visibleCourses = this.coursesHandlerService.removeItem(courseId);
+        console.log('Course has been deleted successfully')
+      }
+    } else throwError(() => "Course wasn't found");
```

Comment:
`throwError` is a creator of observable. If it's not returned from the function and no one subscribes to it - it's useless.

Reply:
You're right, fixed, thank you